src/test/java/io/appform/dropwizard/actors/RabbitmqActorBundleTest.java

```diff
@@ -88,8 +92,31 @@ void testObserverChain() throws Exception {
         Mockito.doReturn(null).when(channel).exchangeDeclare(ArgumentMatchers.anyString(), ArgumentMatchers.anyString(), ArgumentMatchers.anyBoolean(), ArgumentMatchers.anyBoolean(), ArgumentMatchers.any());
         publisher.start();
         publisher.publish(message);
-
         ObserverTestUtil.validateThreadLocal(NamingUtils.queueName(actorConfig.getPrefix(), queueName));
+
+        val delayedQueueName = "queue-delayed-1";
+        val delayedActorConfig = new ActorConfig();
+        delayedActorConfig.setExchange("test-exchange-2");
+        delayedActorConfig.setDelayed(true);
+        delayedActorConfig.setDelayType(DelayType.TTL);
+        val delayedPublisher = new UnmanagedPublisher<>(delayedQueueName, delayedActorConfig, connection, objectMapper);
+        Mockito.doReturn(null).when(channel).exchangeDeclare(ArgumentMatchers.anyString(), ArgumentMatchers.anyString(), ArgumentMatchers.anyBoolean(), ArgumentMatchers.anyBoolean(), ArgumentMatchers.any());
+        delayedPublisher.start();
+
+        verify(channel, times(1)).exchangeDeclare(
+                "test-exchange-2",
+                "direct",
+                true
+        );
+        verify(channel, times(1)).exchangeDeclare(
+                "test-exchange-2_TTL",
+                "direct",
+                true
+        );
+
+        delayedPublisher.publish(message);
+
+        ObserverTestUtil.validateThreadLocal(NamingUtils.queueName(delayedActorConfig.getPrefix(), delayedQueueName));
```

Comment:
move this test case to another method and add common setup things in setup method

Reply:
Okay